pkg/logql/log/pattern/lexer_test.go

```diff
@@ -45,3 +45,30 @@ func toksToStrings(toks []int) []string {
 	}
 	return strings
 }
+
+// TestIdentifierToLower tests that the identifier lexer function converts the identifier to lowercase
+func TestIdentifierToLower(t *testing.T) {
```

Comment:
This is not really testing the right thing.

What this PR is aiming to achieve is to allow queries with keywords being specifiable in either upper- or lower-case characters. This test doesn't test this, nor does it test a valid query.

An appropriate test would be in `pkg/logql/syntax/lex_test.go` which tests a valid query with both upper- and lower-case keywords lexing to the same set of output tokens.

Reply:
Hi @dannykopping,

It's not my intention to bother you but I am stuck, I have been researching all day and trying everything I could  to resolve the error by myself but I was unable to do it.

![image](https://github.com/grafana/loki/assets/99421476/6bf6367d-4241-4758-8b79-b88b7a50a51f)

In the above pic you can see the  ```TestIdentifierToLower``` testing function is successfully passing the test but the thing is in the above reference just for the sake of seeing if the my testing function is valid or not I moved the  ```TestIdentifierToLower``` testing function to ```pkg/logql/log/pattern/lexer_test.go``` same package as ```lexer.go``` in which the main logic of converting to lowercase is defined.

But I know I am not supposed to do that as you've already told that an appropriate test would be in ```pkg/logql/syntax/lex_test.go``` so when I did that I was getting this error you can see in the below pic:

![Test function in lex_test](https://github.com/grafana/loki/assets/99421476/9328ef26-bd96-4b4f-b248-6b3f1eb5a97a)

It would be really helpful if you could tell me what went wrong and how do I fix it, I searched through the Internet all day  but I was unable to fix this, this is the only error that is in our way to get that case-insensitive issue fixed

Reply:
Hi @dannykopping.

Can you please go through above the above comments and tell what do you think about it ?

Reply:
I've been sick the last few days, and will have a look when I've got time 👍 please be patient.

Reply:
Sorry didn't knew that, get well soon